modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala

```diff
@@ -181,6 +181,93 @@ final case class Resolver[F[_]](repos: List[Registry], cache: Option[ResolverCac
   ): F[Either[ResolutionError, Json]] =
     lookupSchemaResult(schemaKey).map(_.map(_.value.schema))
 
+  def lookupSchemasUntil(
+    maxSchemaKey: SchemaKey
+  )(implicit
+    F: Monad[F],
+    L: RegistryLookup[F],
+    C: Clock[F]
+  ): F[Either[ResolutionError, List[Json]]] = {
```

Comment:
Hi @istreeter , `ResolutionError` doesn't tell for which schema the error is. I don't know much about how this function will get called, should I return e.g. `case class SchemaResolutionError(schema: SchemaKey, error: ResolutionError)` instead ?

Reply:
Good question.  Ultimately the return type is used build a "loader iglu error" flavour of bad row.  [Here are the sub-types](https://github.com/snowplow/snowplow-badrows/blob/2.3.0/src/main/scala/com.snowplowanalytics.snowplow.badrows/FailureDetails.scala#L169-L182) we can pick from.

Previously, we have been turning it into a `SchemaListNotFound` bad row:

```
/** Loader couldn't fetch a `SchemaList` - entity necessary for proper shredding with migrations */
final case class SchemaListNotFound(schemaCriterion: SchemaCriterion, error: ClientError) extends LoaderIgluError
```

But after this PR I guess it's more appropriate to return a `IgluError`:

```
/** "Classic" error - loader couldn't find a schema (`ValidationError` should never happen) */
final case class IgluError(schemaKey: SchemaKey, error: ClientError) extends LoaderIgluError
```

So it would be great if your function can return sufficient information that we can construct that failure class.

Reply:
I created [SchemaResolutionError](https://github.com/snowplow/iglu-scala-client/blob/not_use_list/modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala#L487) :

```
case class SchemaResolutionError(schemaKey: SchemaKey, error: ResolutionError)
```
which is more specific than
```
final case class IgluError(schemaKey: SchemaKey, error: ClientError) extends LoaderIgluError
```
and can easily be used to construct it.